webapp/src/errorList.tsx

```diff
@@ -106,132 +110,72 @@ export class ErrorList extends React.Component<ErrorListProps, ErrorListState> {
-    error?: GroupedError;
-    stackframe?: pxsim.StackFrameInfo;
-    location?: pxtc.LocationInfo;
-    blockError?: pxtblockly.BlockDiagnostic;
+    errorGroup: GroupedError;
+    className?: string;
 }
 
 interface ErrorListItemState {
 }
 
-
 class ErrorListItem extends React.Component<ErrorListItemProps, ErrorListItemState> {
     constructor(props: ErrorListItemProps) {
-        super(props)
-
-        this.onErrorListItemClick = this.onErrorListItemClick.bind(this)
+        super(props);
     }
 
     render() {
-        const { error, stackframe, location, blockError } = this.props
-
-        const message = blockError ? lf("{0}", blockError.message)
-            : stackframe ? stackFrameMessageStringWithLineNumber(stackframe, location) :
-                errorMessageStringWithLineNumber(error.error);
-        const errorCount = (stackframe || blockError) ? 1 : error.count;
-
-        return <div className={`item ${stackframe ? 'stackframe' : ''}`} role="button"
-            onClick={!blockError ? this.onErrorListItemClick : undefined}
-            onKeyDown={fireClickOnEnter}
-            aria-label={lf("Go to {0}: {1}", stackframe ? '' : 'error', message)}
-            tabIndex={0}>
-            {message} {(errorCount <= 1) ? null : <div className="ui gray circular label countBubble">{errorCount}</div>}
-        </div>
-    }
-
-    onErrorListItemClick() {
-        const location = this.props.stackframe ? this.props.location : this.props.error.error
-        this.props.revealError(location, this.props.index)
+        const { className, errorGroup } = this.props
+        const error = errorGroup.error;
+
+        const message = error.message;
+
+        return error.stackFrames ? (
+            <div className={className}>
+                <div className="exceptionMessage"
+                    onClick={error.onClick}
+                    onKeyDown={error.onClick ? fireClickOnEnter : undefined}
+                    aria-label={error.message}
+                    tabIndex={0}
+                    role={error.onClick ? "button" : undefined}>
+                    {error.message}
+                </div>
+                <div className="ui selection list">
+                    {(error.stackFrames).map((childErr, index) => {
+                        const errGrp = {error: childErr, count: 1, index: 0};
+                        return <ErrorListItem key={index} index={index} errorGroup={errGrp} className="stackframe"/>
+                    })}
+                </div>
+            </div>
+        ) : (
+            <div className={classList("item", className)} role="button"
+                onClick={error.onClick}
+                onKeyDown={fireClickOnEnter}
+                aria-label={lf("Go to {0}: {1}", error.stackFrames ? '' : 'error', message)}
+                tabIndex={0}>
+                {message} {(errorGroup.count <= 1) ? null : <div className="ui gray circular label countBubble">{errorGroup.count}</div>}
```

Comment:
could we wrap this message in a span or something? feels weird to have text content next to html content